src/actions/query.js

```diff
@@ -81,32 +82,41 @@ export function setHeaders(headers) {
 	};
 }
 
-const SEARCH_COMPONENTS = ['DATASEARCH', 'CATEGORYSEARCH'];
-function msearch(query, orderOfQueries, appendToHits = false) {
+function setSearchId(searchId = null) {
+	return {
+		type: SET_SEARCH_ID,
+		searchId,
+	};
+}
+
+function msearch(query, orderOfQueries, appendToHits = false, isInternalComponent = false) {
 	return (dispatch, getState) => {
 		const {
 			appbaseRef,
 			config,
 			headers,
 			queryListener,
-			selectedValues,
-			componentType,
+			analytics,
 		} = getState();
 
 		let searchHeaders = {};
-		const validComponents = Object.keys(componentType)
-			.filter(item => SEARCH_COMPONENTS.includes(componentType[item]));
 
 		// send search id or term in headers
-		// TODO: implement support for search id
 		if (
 			config.analytics
 			&& config.url === 'https://scalr.api.appbase.io'
-			&& validComponents.length
+			&& !isInternalComponent
```

Comment:
Why are we only filtering internal components here?

Reply:
I'm using this condition to check if the `msearch` was invoked due to an internal component in which case I'm not adding the analytics headers.

Reply:
Refer: https://github.com/appbaseio/reactivecore/pull/15#discussion_r218311514